### PR TITLE
Feature/exe 1886

### PR DESCRIPTION
**Pull request by stefanppetkov**
## Description

Add the following arguments to  `CellCountPlot`:

- `as_frequency = TRUE` shows cell counts as frequencies
- `stack = TRUE` shows a stacked plot of the cell counts/frequencies 

Fixes: EXE-1886

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Existing test for the `CellCountPlot()`function passed.

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked my code and documentation and corrected any misspellings.
- [x] I have run R CMD check on the package and it passes without errors or warnings (notes can be acceptable if motivated)
- [ ] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: R/cell_count_plot.R [hunk 1]
```diff
@@ -59,41 +60,57 @@ CellCountPlot.data.frame <- function(
...
+    nudge_y <- if(as_frequency) max(gg$frequency) / 20 else max(gg$n) / 20
+
+    p <- ggplot(gg, aes(x = .data[[color_by]],
+                        y = if(as_frequency) frequency else n,
+                        fill = .data[[color_by]])) +
       geom_col(position = position_dodge(width = 0.95)) +
       {
         if (show_count) {
-          geom_text(aes(.data[[color_by]], n + nudge_y, label = n))
+          geom_text(aes(label = if(as_frequency) sprintf("%.1f%%", frequency) else n),
```
Comment (ludvigla) [inline R/cell_count_plot.R:102]: The variable `frequency` needs to be added to `globalVariables(...)` in the `aaa.R` script for the package check. You can see the notes here:  https://github.com/PixelgenTechnologies/pixelatorR/actions/runs/10252352635/job/28362538766?pr=52

File: NAMESPACE [hunk 2]
```diff
@@ -239,6 +239,7 @@ importFrom(ggplot2,ggplot_build)
 importFrom(ggplot2,ggplot_gtable)
 importFrom(ggplot2,labs)
 importFrom(ggplot2,position_dodge)
+importFrom(ggplot2,position_stack)
```
Comment (ludvigla) [inline NAMESPACE:242]: I was just about to comment on this :)

There should also be a roxygen2 tag in `pixelatorR-package.R`. The easiest way to add imported functions is to do:

````
usethis::use_import_from("ggplot2", "position_stack")
````

Then run `devtools::document()` to add the import to `NAMESPACE`.

File: R/generics.R [hunk 3]
```diff
@@ -8,8 +8,8 @@ NULL
 #' @rdname as.CellGraphAssay5
 #' @export as.CellGraphAssay5
 as.CellGraphAssay5 <- function (
-  x,
-  ...
+    x,
```
Comment (ludvigla) [inline R/generics.R:11]: Our current style is to use an indentation of 2 for function arguments. No need to change these to 4.

File: R/cell_count_plot.R [hunk 4]
```diff
@@ -30,6 +30,8 @@ CellCountPlot.data.frame <- function(
   color_by,
   show_count = TRUE,
   flip_axes = FALSE,
+  as_frequency = TRUE,
```
Comment (ludvigla) [inline R/cell_count_plot.R:33]: `as_frequency` and `stack` needs to be documented using `@param`.

**Review by ludvigla — CHANGES_REQUESTED** [review]

**Comment by ludvigla** [comment]
Some minor comments that should be addressed before approval.

Also, consider adding a test in `tests/testthat/test-CellCountPlot.R` for the new feature.

File: R/cell_count_plot.R (hunk 1 above)
Comment (stefanppetkov) [reply R/cell_count_plot.R:103]: I have added `frequency` as a global variable.

File: R/cell_count_plot.R (hunk 4 above)
Comment (stefanppetkov) [reply R/cell_count_plot.R]: The parameters are now documented in `generics.R`

File: R/generics.R (hunk 3 above)
Comment (stefanppetkov) [reply R/generics.R]: Fixed!

**Comment by ludvigla** [comment]
Looks great @stefanppetkov! I think this is a very useful feature.

I made a small edit to the documentation of `stack` to highlight that it only has an effect if `group_by` is specified.

**Review by ludvigla — APPROVED** [review]

**Pull request merged**